### DISCUSSION TITLE
High density areas, questions don't show unit number

### DISCUSSION THREAD

**Discussion by bbeattie-phxlabs** [streetcomplete/StreetComplete#5790] [Feedback]
![Screenshot_20240805-161935.png](https://github.com/user-attachments/assets/d2f2860c-8eb3-479c-adb3-31dcdace7c48)

Walking along a street and found it difficult to tell which unit the question was about. Any way of easily enabling that?

**Comment by matkoniecz** [comment]
https://www.openstreetmap.org/node/3994476663 has `addr:unit` tag

**Reply by bradbeattie** [reply]
What's confusing here is that it's the _adjacent_ unit that's been vacant for some time (previous a lamp store). So the pin is misleading, no?

My suggestion is that a little more context would help avoid conflation of adjacent features.

**Reply by bradbeattie** [reply]
Huh. Additionally it highlights the building below (at 1938). Bizarre!

**Reply by matkoniecz** [reply]
https://www.openstreetmap.org/way/83338592 - building was merged with POI which is common but substandard way of mapping and a bit of violation of https://wiki.openstreetmap.org/wiki/One_feature,_one_OSM_element

StreetComplete highlights this area as it is restaurant mapped as an area

**Reply by bradbeattie** [reply]
To the original point, listing the unit number here would have helped disambiguate the feature. But as a working solution, I can ignore the pin location and focus on the purple highlighted feature.

**Reply by bradbeattie** [reply]
![Screenshot_20240921-134247.png](https://github.com/user-attachments/assets/8daf989f-4588-477a-b0fe-c32c9f92b8df)

I'm no longer certain this issue is with the underlying data as I see this issue arise with all features in my area.

**Comment by bradbeattie** [comment]
![Screenshot_20240922-095701](https://github.com/user-attachments/assets/5ad1d50d-8604-4627-9491-fbc4807f8b04)

Another example of mismatched highlighting.

**Reply by matkoniecz** [reply]
It is intentionally highlighting nearby places, for orientation.

Have you expected it to mark only other nearby allotments? Or to not highlight anything in case of question being asked for allotments?

**Reply by mnalis** [reply]
Yes, see _"[StreetComplete FAQ: When I click on quest, several other objects get highlighted, is that a bug?](https://wiki.openstreetmap.org/wiki/StreetComplete/FAQ#When_I_click_on_quest,_several_other_objects_get_highlighted,_is_that_a_bug?)"_ and linked entries for explanation.

In short, if you _ignore all the purple things_, does it become more clear what is being asked about what?
